contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -4016,7 +4015,26 @@ handleBatchLevelStatement(TSqlParser::Batch_level_statementContext *ctx, tsqlSel
 	result->body = list_make1(init);
 	// create PLtsql_stmt_execsql to wrap all query string
 	PLtsql_stmt_execsql *execsql = (PLtsql_stmt_execsql *) makeSQL(ctx);
-	execsql->original_query = pstrdup((makeTsqlExpr(ctx, false))->query);
+	if((ctx->create_or_alter_procedure() && ctx->create_or_alter_procedure()->ALTER()))
+	{
+		int startIndex = ctx->create_or_alter_procedure()->ALTER()->getSymbol()->getStartIndex();
+		std::string result = originalQuery;
+		int endIndex = startIndex + 5;
+		result.replace(startIndex, endIndex - startIndex, "CREATE");
+		execsql->original_query = pstrdup(result.c_str());
```

Comment:
Lets not pstrdup here, instead pass a duplicate copy of the string in `handleBatchLevelStatement`.